deploy.ts

```diff
@@ -44,64 +44,61 @@ let commands: ApplicationCommandsManager
  *
  * @param options Initialization options
  */
-export function init(options: { env: boolean; path?: string }): void
+export function init(options: { env: boolean; path?: string; }): void;
 export function init(options: {
-  publicKey: string
-  token?: string
-  path?: string
-}): void
+  publicKey: string;
+  token?: string;
+  path?: string;
+}): void;
 export function init(options: DeploySlashInitOptions): void {
-  if (client !== undefined) throw new Error('Already initialized')
+  if (client !== undefined) throw new Error('Already initialized');
   if (options.env === true) {
-    options.publicKey = Deno.env.get('PUBLIC_KEY')
-    options.token = Deno.env.get('TOKEN')
+    options.publicKey = Deno.env.get('PUBLIC_KEY');
+    options.token = Deno.env.get('TOKEN');
   }
 
   if (options.publicKey === undefined)
-    throw new Error('Public Key not provided')
+    throw new Error('Public Key not provided');
 
   client = new InteractionsClient({
     token: options.token,
     publicKey: options.publicKey
-  })
+  });
 
-  commands = client.commands
+  commands = client.commands;
 
-  const cb = async (evt: {
-    respondWith: CallableFunction
-    request: Request
-  }): Promise<void> => {
+  const requestHandler = async (request: Request): Promise<Response> => {
     if (options.path !== undefined) {
-      if (new URL(evt.request.url).pathname !== options.path) return
+      if (new URL(request.url).pathname !== options.path) return new Response('Bad Request', {
+        status: 400
+      });
     }
     try {
-      // we have to wrap because there are some weird scope errors
-      const d = await client.verifyFetchEvent({
-        respondWith: (...args: unknown[]) => evt.respondWith(...args),
-        request: evt.request
-      })
-      if (d === false) {
-        await evt.respondWith(
-          new Response('Not Authorized', {
-            status: 400
-          })
-        )
-        return
+      const interaction = await client.verifyFetchEvent({ request, respondWith: (res: Response) => { return res; } });
+      if (!interaction) {
+        // ... unauthorized
+        return new Response('Not Authorized', {
+          status: 400
+        });
       }
-
-      if (d.type === InteractionType.PING) {
-        await d.respond({ type: InteractionResponseType.PONG })
-        client.emit('ping')
-        return
+      if (interaction.type === InteractionType.PING) {
+        await interaction.respond({ type: InteractionResponseType.PONG });
+        await client.emit('ping');
       }
 
-      await client._process(d)
-    } catch (e) {
-      await client.emit('interactionError', e as Error)
+      await client._process(interaction);
+      return new Response('Error', {
+        status: 500
+      });
```

Comment:
Why is this returning error (after processing interaction)? It should be 200 as OK or 203 as empty response

Reply:
I don't think so, it only gets run if "await client.verifyFetchEvent({ request, respondWith: (res: Response) => { return res; } });" fails to return anything, and is there to avoid TS problems.